### PR TITLE
RMB-814: PostgresClient.selectStream futurisation

### PR DESCRIPTION
**Pull request by julianladisch**

### CODE REVIEW

**Review by adamdickmeiss — APPROVED** [review]

File: domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java [hunk 1]
```diff
@@ -866,4 +867,21 @@ static String durationMsg(String descriptionKey, String sql, long startNanoTime)
...
     return streamGet(table, clazz, PostgresClient.DEFAULT_JSONB_FIELD_NAME, wrapper, false, null, null, streamHandler);
   }
+
+  /**
+   * Get a stream of the results of the {@code sql} query.
+   *
+   * @param params arguments for {@code $} placeholders in {@code sql}
+   * @param chunkSize cursor fetch size
+   */
+  public Future<RowStream<Row>> selectStream(String sql, Tuple params, int chunkSize) {
```
Comment (hjiebsco) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java:877]: Will there be a futurized version of `selectStream` in `PostgresClient`?

**Comment by julianladisch** [comment]
No.
When selectStream returns a RowStream the connection is still open.
You always need a connection and should use Conn if you want to use futurised methods:

```
PostgresClient.withConn(conn -> {
  return conn.selectStream(...)
  .compose ...
});
```

**Comment by hjiebsco** [comment]
> No.
> When selectStream returns a RowStream the connection is still open.
> You always need a connection and should use Conn if you want to use futurised methods:
> 
> ```
> PostgresClient.withConn(conn -> {
>   return conn.selectStream(...)
>   .compose ...
> });
> ```

So consumer has to use `PostgresClient` for some methods but `Conn` for others? Is it possible to simplify it by just exposing `PostgresClient` but not `Conn`?

File: domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java (hunk 1 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java]: I've added it.

**Comment by julianladisch** [comment]
I've changed the implementation with better futurisation.
PostgresClient.selectStream should be used if RMB should create the connection and the transaction.
Conn.selectStream should be used if the query should run on an existing connection in an existing transaction.

**Review by hjiebsco — APPROVED** [review]

**Pull request merged**